numpy/linalg/_linalg.pyi

```diff
@@ -480,3 +487,27 @@ def matmul(
     x1: _ArrayLikeComplex_co,
     x2: _ArrayLikeComplex_co,
 ) -> NDArray[complexfloating]: ...
+
+@overload
+def geneigvals(a: _ArrayLikeInt_co, b: _ArrayLikeInt_co) -> NDArray[float64]: ...
+@overload
+def geneigvals(a: _ArrayLikeFloat_co, b: _ArrayLikeFloat_co) -> NDArray[floating]: ...
+@overload
+def geneigvals(a: _ArrayLikeComplex_co, b: _ArrayLikeComplex_co) -> NDArray[complexfloating]: ...
+
+@overload
+def geneig(
+    a: _ArrayLikeInt_co,
+    b: _ArrayLikeInt_co,
+) -> PolyEigResult: ...
+@overload
+def geneig(
+    a: _ArrayLikeFloat_co,
+    b: _ArrayLikeFloat_co,
+) -> PolyEigResult: ...
```

Comment:
there's no need for overloads if the return types are the same

```suggestion
def geneig(a: _ArrayLikeFloat_co, b: _ArrayLikeFloat_co) -> PolyEigResult: ...
```

Another option would be to keep these overloads, and turn `PolyEigResult` in a [generic type](https://typing.python.org/en/latest/reference/generics.html), so that you can write things like `PolyEigResult[np.float64]`.

Reply:
This definition has the same issue with parameter names as the one above.